server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/BinaryValuesSource.java

```diff
@@ -162,7 +162,7 @@ public void collect(int doc, long bucket) throws IOException {
     }
 
     @Override
-    LeafBucketCollector getLeafCollector(Comparable value, LeafReaderContext context, LeafBucketCollector next) {
+    LeafBucketCollector getLeafCollector(Comparable<?> value, LeafReaderContext context, LeafBucketCollector next) {
```

Comment:
Same thing here, assuming we update `SingleDimensionValuesSource` we should make this `Comparable<BytesRef>`